### ISSUE TITLE
Playlist flag crashing Windows and Android when InPrivate Window is opened.

### ISSUE THREAD

**Issue by Emi-TheDhamphirInLoveUnderTheFrozenStar** [brave/brave-browser#30037]
## Description 

Hi,
After Nightly 1.52.72 (April 25th), both Windows and Android browsers have been instant crashing when opening InPrivate window and the Playlist flag is enabled.

On Desktop, I even tested on a completely new User Data with only the Playlist flag enabled to discard there could be another variable for the crash; it still crashed.

On Android I noticed it was crashing as well, and just disabled the flag and no more crashes

@sangwoo108

## Steps to Reproduce

   1. Enable `brave://flags/#playlist`
   2. restart and open InPrivate window
   3. should crash instantly.

## Desktop Brave version:

1.53.2 Chromium: 113.0.5672.53 (Official Build) nightly (64-bit) 
Windows 11 Version 21H2 (Build 22000.1817)

## Android Device details:
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): Phone
   - Android version: 13

**Comment by sangwoo108** [comment]
Yeah, we're not going to support Playlist on InPrivate Window for security reason, but didn't handle it properly. Thanks for reporting.

**Comment by Emi-TheDhamphirInLoveUnderTheFrozenStar** [comment]
Will the PR also disable the button in sidebar on the InPrivate window, so it can't be clicked on?

I always had a feeling Playlist was meant to work only in Normal windows, because of security reasons and for everything Playlist has to do to download files and have access to those files to play them and all that, so I never put too much thought on it, and just learned not to click the button since the day it was available on Desktop, but sometimes I accidentally clicked on the playlist button, causing the browser to crash. So I better ask to know once and for all.

Anyway, thanks for your work! Playlist will be a cool and useful on Android for sure.

**Comment by Emi-TheDhamphirInLoveUnderTheFrozenStar** [comment]
@sangwoo108 1.53.5 is causing high CPU when a Private window is opened, as soon as the window is closed CPU levels go back to normal and if you disable the Playlist flag the CPU levels are normal in Private windows.

Task manager shows high CPU is from `Private Extension: Brave` (and any other extension allowed to to run in Private mode), and any opened tab, even if it is the Private NTP.

Brave process, any normal window tab or extension processes, the GPU process or any other Utility Service, work fine with not high CPU usage.

This is how Brave looks on my side [with System Informer]. New User Data and only Playlist flag enabled:

![brave_2023-04-28_11-24-32](https://user-images.githubusercontent.com/122518587/235213865-fedc9fb4-1268-447f-8ebf-aca7c4aab353.gif)

**Comment by sangwoo108** [comment]
@rebron , this should be uplifted to both 52 and 51, shouldn't it?

**Comment by sangwoo108** [comment]
> https://github.com/brave/brave-browser/issues/30037#issuecomment-1527886281

Thanks @Emi-TheDhamphirInLoveUnderTheFrozenStar . That's a really nice find. But I'm a little bit surprised as Playlist doesn't use any extension facilities AFAIK.

**Comment by bsclifton** [comment]
@sangwoo108 yes please!

**Comment by sangwoo108** [comment]
> https://github.com/brave/brave-browser/issues/30037#issuecomment-1529956456

Oh, this was caused by a patch merged on 1.52.x, so I guess we don't need to uplift it to 1.51.x.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.53.50 Chromium: 114.0.5735.26 (Official Build) nightly (64-bit)
-- | --
Revision | 7075cbb66f0542ac3e01ddfde6b813e7d61118a5-refs/branch-heads/5735@{#454}
OS | Windows 11 Version 22H2 (Build 22621.1702)
```

Verification notes for the above can be found via https://github.com/brave/brave-core/pull/18289 as we decided not to uplift the above as `Play

**Comment by kjozwiak** [comment]
As discussed with @deeppandya via https://bravesoftware.slack.com/archives/C03MMJ4CHEZ/p1686260940735599, this issue will be added into the `master` playlist PR. This way the above won't be missed and will be verified. Reasoning is that the above code landed in `1.53.x` so the issue should stay in `1.53.x` and shouldn't be moved into other milestones. Removing `OS/Android` so it doesn't appear in our list for `1.53.x` verifications.
